internal/pkg/siderolink/manager.go

```diff
@@ -265,17 +265,36 @@ func (manager *Manager) Run(
 	trustdPort string,
 	logServerPort string,
 ) error {
-	eg, groupCtx := errgroup.WithContext(ctx)
+	eg, ctx := errgroup.WithContext(ctx)
 
-	if err := manager.startWireguard(groupCtx, eg, manager.serverAddr); err != nil {
+	if err := manager.runWireguard(ctx, eg, listenHost, eventSinkPort, trustdPort, logServerPort); err != nil {
 		return err
 	}
 
-	eg.Go(func() error { return manager.pollWireguardPeers(groupCtx) })
+	return eg.Wait()
+}
 
-	eg.Go(func() error {
-		return manager.provisionServer.runCleanup(groupCtx)
-	})
+func (manager *Manager) runWireguard(ctx context.Context, eg *errgroup.Group, listenHost, eventSinkPort, trustdPort, logServerPort string) error {
+	wireguardCtx, wireguardCancel := context.WithCancel(context.Background())
+	defer wireguardCancel()
+
+	if err := manager.startWireguard(wireguardCtx, eg, manager.serverAddr); err != nil { //nolint:contextcheck
```

Comment:
what does shutdown `startWireguard` now?

Reply:
`startWireguard` is now receiving a cancellable context and a different error group, so we can wait for all wireguard dependent services(incuding logServer) to finish separately before attempting to tore down wireguard connection.